src/marten/db/query/sql/concerns/sanitizer.cr

```diff
@@ -0,0 +1,67 @@
+module Marten
+  module DB
+    module Query
+      module SQL
+        module Sanitizer
+          private NAMED_PARAMETER_RE        = /(:?):([a-zA-Z]\w*)/
+          private POSITIONAL_PARAMETER_CHAR = '?'
+          private POSITIONAL_PARAMETER_RE   = /#{"\\" + POSITIONAL_PARAMETER_CHAR}/
+
+          private def sanitize_positional_parameters(
+            query : String,
+            params : Array(::DB::Any),
+            connection : Connection::Base? = nil
```

Comment:
The `connection` argument should be mandatory here. Otherwise, we can't be sure that the generated parameters will work for the actual database at hand. Is there a rationale for making it optional? From what I can see we have access to a connection object in all places where this module is used.

Reply:
For the raw queries this function will replace the placeholders with a `%s` instead of database specific prepared statement parameter.

The reason for this is, that [`where_clause_and_parameters`](https://github.com/martenframework/marten/blob/main/src/marten/db/query/sql/query.cr#L1117) will format the prepared statement afterwards. For Sqlite and Mysql/Mariadb there won't be a problem, but Postgres uses numbered prepared statement parameters which will potentially yield a wrong order because `Query `does not know about the order of the `RawPredicateNode#to_sql` order.

I'd happy if you have any insights for this ordering problem

Reply:
That makes sense. Let's keep it that way, I don't think we could find a way to go around the way the parameters are handled in the main `Marten::DB::Query::SQL::Query` abstraction when building the where clause and parameters.